text/0000-export-report.md

```diff
@@ -0,0 +1,84 @@
+# Meta
+[meta]: #meta
+- Name: Export Report
+- Start Date: (fill in today's date: 2020-04-01)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+The lifecycle export stage should optionally write a machine parsable metadata "report" about the resulting exported image(s).
+
+# Motivation
+[motivation]: #motivation
+
+The logging output of the lifecycle export stage displays the built image's digest or image id. However, this information is not easily available for a platform that needs to utilize the resulting image from a build. 
+
+Without this information, a platform will need to resort to suboptimal mechanisms such as parsing the exported output or reading the most recently tagged image on the registry. 
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The lifecycle exporter/creator binary will accept an optional flag `report` which will provides a path to `report.toml`. When the flag is supplied the export stage will write a toml representation of the built image for platform consumption. When the flag is not provided it will default to writing report.toml in the working directory which by convention is `/layers`.
+
+Here is an overview of the suggested schema for report.toml:
+
+On a build published to a registry:
+```
+[image]
+tags = ["index.docker.io/image/name:latest", "index.docker.io/image/name:other-tag"]
+digest = "sha256:c8be1b8f4d60d99c281fc2db75e0f56df42a83ad2f0b091621ce19357e19d853"
+```
+
+On a build exported to the docker daemon:
+```
+[image]
+tags = ["index.docker.io/image/name:latest", "index.docker.io/image/name:other-tag"]
+image_id = "9c0e1895b90f"
+```
+
+By representing this in toml, we can extend this information with additional metadata when needed. A general philosophy for report.toml is that it should contain information that is not accessible elsewhere. For example, the bill of materials should not be included as it is available as an image label and would be unnecessarily duplicated in report.toml. 
```

Comment:
Is there any decent tooling for extracting `toml` keys?  Something like `jq` for `toml`?

Suppose what I wanted in a particular result file (e.g. in Tekton) was the image digest, what is the guidance for extracting particular keys?

Reply:
You could use [yj]() with jq. `yj -t | jq -r .image.digest`

Reply:
This is what we've been doing in the heroku buildpacks.

Reply:
Cool, TIL.